client/chat/channel.jsx

```diff
@@ -237,12 +237,12 @@ class Channel extends React.Component {
     isScrolledUp: false,
   }
 
-  componentWillUpdate(nextProps, nextState) {
+  componentDidUpdate(prevProps) {
     const insertingAtTop =
-      nextProps.channel !== this.props.channel &&
-      this.props.channel.messages.size > 0 &&
-      nextProps.channel.messages.size > this.props.channel.messages.size &&
-      nextProps.channel.messages.get(0) !== this.props.channel.messages.get(0)
+      prevProps.channel !== this.props.channel &&
+      prevProps.channel.messages.size > 0 &&
+      this.props.channel.messages.size > prevProps.channel.messages.size &&
+      this.props.channel.messages.get(0) !== prevProps.channel.messages.get(0)
     this.messageList.setInsertingAtTop(insertingAtTop)
```

Comment:
I thnk `componentDidUpdate` is too late to do this. In any case, this PR should contain both ScrollableContent changes *and* these, because the right place to do this depends on both changes.

Reply:
I tried using `getSnapshotBeforeUpdate()` instead and I couldn't make it work. It could probably be done, but I tested it as much as I could with `componentDidUpdate()` and everything works as far as I can see.